lib/fog/vsphere/models/compute/volume.rb

```diff
@@ -49,7 +49,8 @@ def save
           requires :server_id, :size, :datastore
 
           if unit_number.nil?
-            used_unit_numbers = server.volumes.map { |volume| volume.unit_number }
+            # Vsphere maps unit_numbers 7 and greater to a higher SCSI ID since the pvscsi driver reserves SCSI ID 7
+            used_unit_numbers = server.volumes.map { |volume| volume.unit_number < 7 ? volume.unit_number : volume.unit_number - 1 }
```

Comment:
If it is to a higher number as in your comment why are you decreasing by one instead? Shouldn't you be increasing by one if number is equal 7 or greater?